test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1981,7 +1981,9 @@ UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_am_nbx_rndv_memtype_disable_zcopy);
 #ifdef ENABLE_STATS
 class test_ucp_am_nbx_rndv_ppln : public test_ucp_am_nbx_rndv {
 public:
-    test_ucp_am_nbx_rndv_ppln() : m_mem_type(UCS_MEMORY_TYPE_HOST) {}
+    test_ucp_am_nbx_rndv_ppln() : m_mem_type(UCS_MEMORY_TYPE_HOST) {
+        modify_config("PROTO_ENABLE", "y");
```

Comment:
it is not needed, the test is skipped for protov1  (see `init` on line 1990)

Reply:
The problem is the base class `test_ucp_am_base` constructor, that treats any variant value as proto v1. To ignore that behaviour of the base class I revert setting back to proto v2.
```C
    test_ucp_am_base()
    {
        if (get_variant_value()) {
            modify_config("PROTO_ENABLE", "n");
        }
    }
```
This seems to be the least intrusive way to fix it, I was afraid of changing that base class due to many subclasses reusing it

Reply:
so, maybe just skip wrong variants and add missed if any?

Reply:
It's not about skipping wrong variants.
Look, as a part of this commit I've added a new variant (error handling).
But, the base class misinterprets this variant - it treats **any non-zero variant of level 0 as proto_v1** which is wrong IMO (see code above). It makes all my error-handling tests = proto_v1 tests, which are then skipped.
So I see the following options to fix it:
1. The easiest is what's implemented here: just always force proto_v2, and ignore base class misinterpretation of new variant
2. use level 0 variant for proto_v1 (to conform the base class interpretation), and add level 1 variant for error handling. Then obviously we skip all proto_v1 tests.
3. Change impl of the base class ctor, and all the subclasses using it
Any other ideas?
I thought option 1 is the best choice here

Reply:
The code above looks weird to me, from first look it seems like something temporary and hacky to pass CI on transition from proto_v1 to proto_v2.
If we need some derived tests working only with proto_v1 and others with proto_v2, I would suggest to change inheritance correspondingly.

Reply:
this class defines its own variants, so should modify base class probably to not set it when it is not needed

Reply:
You can introduce enum.
```
enum {
    VARIANT_DEFAULT = 0,
    VARIANT_PROTO_V1,
    VARIANT_ERR_HANDLING
};

class test_ucp_am_base : public ucp_test {
...
    test_ucp_am_base()
    {
        if (get_variant_value() == VARIANT_PROTO_V1) {
            modify_config("PROTO_ENABLE", "n");
        }
    }

    static void get_test_variants(variant_vec_t &variants)
    {
        add_variant_with_value(variants, UCP_FEATURE_AM, VARIANT_DEFAULT, "");
        if (!RUNNING_ON_VALGRIND) {
            add_variant_with_value(variants, UCP_FEATURE_AM, VARIANT_PROTO_V1, "proto_v1");
        }
    }
...
};

...

class test_ucp_am_nbx_rndv_ppln : public test_ucp_am_nbx_rndv {
...
    static void get_test_variants(variant_vec_t &variants)
    {
        if (!mem_buffer::is_gpu_supported()) {
            return;
        }

        add_variant_with_value(variants, UCP_FEATURE_AM, VARIANT_DEFAULT , "");
        add_variant_with_value(variants, UCP_FEATURE_AM, VARIANT_ERR_HANDLING, "error-handling");
    }

    get_err_mode() {
        return (get_variant_value() == VARIANT_ERR_HANDLING) ? UCP_ERR_HANDLING_MODE_PEER : UCP_ERR_HANDLING_MODE_NONE;
    }
```

Reply:
Thanks for the code, I agree that this is the right way of refactoring

Reply:
After reflecting a bit more on this, I realised that enum approach might not be the best candidate..
It allows to choose just a single option (default, proto_v1, errh) for level 0, but there are tests in this file that use both combinations: proto_v1 AND errh: e.g. `test_ucp_am_nbx_dts` So this enum does not really map well with this. TBH I would stay with initial approach that is one-liner: ignore base class protocol setting and treat it as error-handling mode

Reply:
I would recommend to use enum anyway. The code in the base class (which was partially written by) is difficult to extend. A good practice is to add specific value using `add_variant_with_value`, and compare with it when calling `get_variant_value`.

And if it is required to multiply the test cases it should be done like it already done:
```
static void get_test_variants(variant_vec_t &variants)
{
    add_variant_values(variants, base_class_name::get_test_variants, VARIANT_DEFAULT);
    add_variant_values(variants, base_class_name::get_test_variants, VARIANT_ERR_HANDLING, "error-handling");
}

get_err_mode() {
    return (get_variant_value(<index>) == VARIANT_ERR_HANDLING) ? UCP_ERR_HANDLING_MODE_PEER : UCP_ERR_HANDLING_MODE_NONE;
}
```

Reply:
side note. I think the